tests/lib/eslint.js

```diff
@@ -723,6 +723,137 @@ describe("eslint", function() {
             var messages = eslint.verify("0", config);
             assert.equal(messages[0].message, "my message testing!");
         });
+
+        it("should allow template parameter with inner whitespace", function() {
+            eslint.reset();
+            eslint.defineRule("test-rule", function(context) {
+                return {
+                    "Literal": function(node) {
+                        context.report(node, "message {{parameter name}}", {
+                            "parameter name": "yay!"
+                        });
+                    }
+                };
+            });
+
+            var config = { rules: {} };
+            config.rules["test-rule"] = 1;
+
+            var messages = eslint.verify("0", config);
+            assert.equal(messages[0].message, "message yay!");
+        });
+
+        it("should allow template parameter with non-identifier characters", function() {
+            eslint.reset();
+            eslint.defineRule("test-rule", function(context) {
+                return {
+                    "Literal": function(node) {
+                        context.report(node, "message {{parameter-name}}", {
+                            "parameter-name": "yay!"
+                        });
+                    }
+                };
+            });
+
+            var config = { rules: {} };
+            config.rules["test-rule"] = 1;
+
+            var messages = eslint.verify("0", config);
+            assert.equal(messages[0].message, "message yay!");
+        });
+
+        it("should ignore template parameter with no specified value", function() {
+            eslint.reset();
+            eslint.defineRule("test-rule", function(context) {
+                return {
+                    "Literal": function(node) {
+                        context.report(node, "message {{parameter}}", {});
+                    }
+                };
+            });
+
+            var config = { rules: {} };
+            config.rules["test-rule"] = 1;
+
+            var messages = eslint.verify("0", config);
+            assert.equal(messages[0].message, "message {{parameter}}");
+        });
+
```

Comment:
NOTE: All tests before this point pass even with old code. These are the tests which document previously undocumented assumptions. All tests after this point fail with the old code.